### PR TITLE
Add League Versus Match card

### PR DESCRIPTION
**Pull request by CarolineMorton12**
### Trello

https://trello.com/c/REEfms91/61-frontend-int-versus-match-card

### Description

made a `MatchCard` with a cute green background to the winning score

### CODE REVIEW

File: packages/frontend/src/components/match_card.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
+export function MatchCard(props: { score: Record<string, number> }) {
```
Comment (WilfSilver) [inline packages/frontend/src/components/match_card.tsx:1]: Could we change the name to `VersusMatchCard` (along with the filename) so it doesn't collide with 2 other match cards

File: packages/frontend/src/components/match_card.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  const firstValue = props.score[firstKey];
+  const secondValue = props.score[secondKey];
+  return (
+    <div className="bg-gray-400 text-center w-min rounded-lg text-white overflow-hidden">
+      <div className="flex justify-between whitespace-nowrap text-xl font-bold items-center">
+        <p className="m-2">{firstKey} </p>
+        <p
+          className="ml-2 p-2"
+          style={{
+            backgroundColor: firstValue >= secondValue ? "green" : undefined,
```
Comment (WilfSilver) [inline packages/frontend/src/components/match_card.tsx:12]: This should be using tailwind with `bg-p-green` type so its using our primary green colour

File: packages/frontend/src/components/match_card.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        <div className="bg-white h-1" />
+        <p className="leading-[0]">vs</p>
+        <div className="bg-white h-1" />
+      </div>
+      <div className="flex justify-between whitespace-nowrap text-xl font-bold items-center">
+        <p className="m-2">{secondKey} </p>
+        <p
+          className="ml-2 p-2"
+          style={{
+            backgroundColor: firstValue <= secondValue ? "green" : undefined,
```
Comment (WilfSilver) [inline packages/frontend/src/components/match_card.tsx:28]: Same as above.

Also as a side note, how do you feel about always putting the winner on the top?

File: packages/frontend/src/components/match_card.tsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+          }}
+        >
+          {secondValue}
+        </p>
+      </div>
```
Comment (WilfSilver) [inline packages/frontend/src/components/match_card.tsx:33]: This feels a bit repetitive, might be worth looking into using loops or something (if not don't worry)

**Review by WilfSilver — CHANGES_REQUESTED** [review]
Looking good!

As a side note, this will not be merged without unit testing. Hopefully we will go over this this afternoon

**Review by WilfSilver — COMMENTED** [review]
LGTM - will wait for testing though

File: packages/frontend/src/components/versus_match_card.tsx [hunk 5]
```diff
@@ -0,0 +1,30 @@
+export function VersusMatchCard(props: { score: Record<string, number> }) {
+  const [firstKey, secondKey] = Object.keys(props.score);
+  const firstValue = props.score[firstKey];
+  const secondValue = props.score[secondKey];
+  return (
+    <div className="bg-gray-400 text-center w-min rounded-lg text-white overflow-hidden">
+      <div
+        className={`${
+          secondValue <= firstValue ? "bg-p-green-200" : ""
```
Comment (WilfSilver) [inline packages/frontend/src/components/versus_match_card.tsx:9]: As you can see with the coverage (and in the files changed bit) this branch is not covered, looking at your tests, you never check if the second person is the winner and not the first

File: packages/frontend/tests/versus_match_card.test.tsx [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+
+    const div1 = matchCard.children.item(0) as HTMLElement;
+    const div2 = matchCard.children.item(2) as HTMLElement;
+
+    expect(div1).not.toBeNull();
+    expect(div2).not.toBeNull();
+
+    const p11 = div1.children.item(0);
+    expect(p11).not.toBeNull();
+    expect(p11!.textContent).toBe("Lara");
```
Comment (WilfSilver) [inline packages/frontend/tests/versus_match_card.test.tsx:29]: Linting error. As you are not using vscode, please make sure to run `yarn lint:fix` before pushing and fixing any issues.

In this case it can just be swapped to a `?` as we don't care about the type.

Also as a subnote, we prefer tests failing instead of crashing! The null assertion would crash the tests (meaning no other tests can run) which is really hard to debug

```suggestion
    expect(p11?.textContent).toBe("Lara");
```

File: packages/frontend/tests/versus_match_card.test.tsx [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+    const p12 = div1.children.item(1);
+    expect(p12).not.toBeNull();
+    expect(p12!.textContent).toBe("10");
```
Comment (WilfSilver) [inline packages/frontend/tests/versus_match_card.test.tsx:32]:
```suggestion
    expect(p12?.textContent).toBe("10");
```

File: packages/frontend/tests/versus_match_card.test.tsx [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+
+    const p21 = div2.children.item(0);
+    expect(p21).not.toBeNull();
+    expect(p21!.textContent).toBe("Hugo");
```
Comment (WilfSilver) [inline packages/frontend/tests/versus_match_card.test.tsx:36]:
```suggestion
    expect(p21?.textContent).toBe("Hugo");
```

File: packages/frontend/tests/versus_match_card.test.tsx [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+    const p22 = div2.children.item(1);
+    expect(p22).not.toBeNull();
+    expect(p22!.textContent).toBe("1");
```
Comment (WilfSilver) [inline packages/frontend/tests/versus_match_card.test.tsx:39]:
```suggestion
    expect(p22?.textContent).toBe("1");
```

File: packages/frontend/tests/versus_match_card.test.tsx [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+
+    const matchCard = component.container.children.item(0) as HTMLElement;
+    expect(matchCard).not.toBeNull();
+
+    const div1 = matchCard.children.item(0) as HTMLElement;
+    const div2 = matchCard.children.item(2) as HTMLElement;
+    expect(div1).not.toBeNull();
+    expect(div2).not.toBeNull();
+
+    expect(div1!.classList.contains("bg-p-green-200")).toBe(true);
```
Comment (WilfSilver) [inline packages/frontend/tests/versus_match_card.test.tsx:59]: `!` can be removed (check linter)

File: packages/frontend/tests/versus_match_card.test.tsx [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+    expect(div2!.classList.contains("bg-p-green-200")).toBe(false);
```
Comment (WilfSilver) [inline packages/frontend/tests/versus_match_card.test.tsx:60]: `!` can be removed - you cast it above to `HTMLElement` which means it can't be `undefined` (though technically it can you are just telling typescript that it won't)

File: packages/frontend/tests/versus_match_card.test.tsx [hunk 12]
```diff
@@ -0,0 +1,83 @@
...
+
+    const matchCard = component.container.children.item(0) as HTMLElement;
+    expect(matchCard).not.toBeNull();
+
+    const div1 = matchCard.children.item(0) as HTMLElement;
+    const div2 = matchCard.children.item(2) as HTMLElement;
+    expect(div1).not.toBeNull();
+    expect(div2).not.toBeNull();
+
+    expect(div1!.classList.contains("bg-p-green-200")).toBe(true);
```
Comment (WilfSilver) [inline packages/frontend/tests/versus_match_card.test.tsx:80]:
```suggestion
    expect(div1.classList.contains("bg-p-green-200")).toBe(true);
```

File: packages/frontend/tests/versus_match_card.test.tsx [hunk 13]
```diff
@@ -0,0 +1,83 @@
...
+    expect(div2!.classList.contains("bg-p-green-200")).toBe(true);
```
Comment (WilfSilver) [inline packages/frontend/tests/versus_match_card.test.tsx:81]:
```suggestion
    expect(div2.classList.contains("bg-p-green-200")).toBe(true);
```

**Review by WilfSilver — CHANGES_REQUESTED** [review]
As mentioned in the comments, as you are not using our VSCode setup, please make sure to run `yarn lint:fix` before you commit to fix any linting issues (you may have to manually intervene with some

**Review by WilfSilver — APPROVED** [review]
LGTM
